homeassistant/components/deconz/light.py

```diff
@@ -158,111 +170,139 @@ def color_mode(self) -> str:
         return color_mode
 
     @property
-    def brightness(self):
+    def brightness(self) -> int | None:
         """Return the brightness of this light between 0..255."""
-        return self._device.brightness
+        if self._device.brightness:
+            return int(self._device.brightness)
+        return None
 
     @property
-    def color_temp(self):
+    def color_temp(self) -> int:
         """Return the CT color value."""
-        return self._device.color_temp
+        return int(self._device.color_temp)
 
     @property
-    def hs_color(self) -> tuple:
+    def hs_color(self) -> tuple[float, float]:
         """Return the hs color value."""
-        return (self._device.hue / 65535 * 360, self._device.saturation / 255 * 100)
+        return (
+            int(self._device.hue) / 65535 * 360,
+            int(self._device.saturation) / 255 * 100,
+        )
 
     @property
-    def xy_color(self) -> tuple | None:
+    def xy_color(self) -> tuple[float, float] | None:
         """Return the XY color value."""
-        return self._device.xy
+        if self._device.xy is not None:
+            x_value: float = self._device.xy[0]
+            y_value: float = self._device.xy[1]
+            return (x_value, y_value)
+        return None
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if light is on."""
-        return self._device.state
+        return bool(self._device.state)
 
-    async def async_turn_on(self, **kwargs):
+    async def async_turn_on(
+        self, **kwargs: bool | float | int | str | tuple[float, float]
+    ) -> None:
         """Turn on light."""
-        data = {"on": True}
+        data: dict[str, bool | float | int | str | tuple[float, float]] = {"on": True}
 
-        if ATTR_BRIGHTNESS in kwargs:
-            data["brightness"] = kwargs[ATTR_BRIGHTNESS]
+        if attr_brightness := kwargs.get(ATTR_BRIGHTNESS):
+            assert isinstance(attr_brightness, int)
```

Comment:
avoid using asserts

Reply:
I got suggested to use either type casting or to use # type: ignore[no-any-return] until library is utilised for typing (https://github.com/home-assistant/core/pull/59610#discussion_r748765037)

Reply:
So, IMHO, as you control the library, fix that first instead. This is working the wrong way around.